aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1026,7 +1026,7 @@ Tensor alias_with_sizes_and_strides(
     self_ = Tensor(std::move(impl));
   } else {
     auto impl = c10::make_intrusive<TensorImpl>(
-        Storage(self.storage()), self.key_set(), self.dtype());
+        Storage(self.storage()), self.key_set(), self.dtype(), /*is_view=*/true);
```

Comment:
I find it hard to believe that this is the only function that creates views. Even within this file, there are five occurrences of `git grep "Storage(self.storage()"` (and I'm not even sure those are the only occurrences)

Reply:
This is a great opportunity for a staging PR before this one. If you have concluded that there is *no other way* to do what you want to do besides forcing constructors of TensorImpl to declare if they are a view or not, a good, easy prep commit would be to just do that (but maybe not do anything with the information?) and devise a way to validate that you'd gotten all of the necessary sites.